### PR TITLE
Parse URL to fix section handling

### LINKED ISSUE #93: Search: resource links to 3rd level pages are broken
### Steps to Reproduce

1. Search  [“Water & Oceans”](https://ocwnext.odl.mit.edu/search/?q=%22water%20%26%20oceans%22&type=resourcefile) 
2. Click on the first result, https://ocwnext.odl.mit.edu/courses/ec-719-d-lab-water-climate-change-and-health-spring-2019/sections/water-oceans/

### Expected Behavior

- The link should be to https://ocwnext.odl.mit.edu/courses/ec-719-d-lab-water-climate-change-and-health-spring-2019/sections/additional-resources/water-oceans/

### Actual Behavior

- We get an S3 error message, equivalent to a 404. 

### Related Issues

We seem to be missing part of the path, which is in the `section_slug` field of the search results:

```
{
run_slug: "ec-719-d-lab-water-climate-change-and-health-spring-2019",
section_slug: "additional-resources",
short_url: "water-oceans"
}
```

### PR DESCRIPTION
**Pull request by noisecapella**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Fixes #93 

#### What's this PR do?
Instead of using `short_url`, this parses `url` which is the legacy url in order to get the section parts. Ideally this would be done in ocw-data-parser but at this point it seems simpler to fix it here.

#### How should this be manually tested?
Search for "water oceans" with the quotes. Click on "resources". Hover over the link in the first result. It should be something like `http://localhost:8063/courses/ec-719-d-lab-water-climate-change-and-health-spring-2019/sections/additional-resources/water-oceans/`, depending on your open-discussions instance

### CODE REVIEW

**Comment by pdpinch** [comment]
Should we still fix this in ocw-data-parser?

If we don’t fix it in ocw-data-parser, does that mean we will be importing incorrect data into ocw-studio?

> Ideally this would be done in ocw-data-parser but at this point it seems simpler to fix it here.

**Comment by noisecapella** [comment]
For the second question, no, we are using correct urls in ocw-studio already. We are updating these links in ocw-to-hugo when we export to markdown. ocw-studio imports from the markdown, not from the parsed JSON, so it is dealing with links which are already updated. 

For the first question, possibly but it's a cost/benefit question. open-discussions imports from the plone JSON, using ocw-data-parser to convert the plone JSON files to a single parsed JSON. That means the `ContentFile` objects have incorrect urls in the search results, which this PR fixes in Javascript (which we are already doing for other links). This is all needlessly complicated but if we are getting rid of ocw-data-parser and ocw-to-hugo anyway in the near future it might not be worth trying to fix it at that level.

**Comment by gumaerc** [comment]
Just a heads up, we might want to move the code in this PR over to `ocw-hugo-themes` as the search app will live there going forward.  https://github.com/mitodl/ocw-www/pull/101 will be merged as part of the cutover and will remove the search app from this repo.

File: src/js/lib/search.js [hunk 1]
```diff
@@ -460,11 +460,34 @@ export const getCourseUrl = result => {
     null
 }
 
+export const getSectionUrl = result => {
+  let url = result.url
+  // a small number of urls still start with the legacy site prefix
+  const legacyPrefix = "http://ocw.mit.edu"
```
Comment (aliceriot) [inline src/js/lib/search.js:466]: do we have to worry about `http/https` here? if so we might want to 

```js
/https?:\/\/ocw.mit.edu/
```

**Review by aliceriot — CHANGES_REQUESTED** [review]
just had one question

File: src/js/lib/search.js (hunk 1 above)
Comment (noisecapella) [reply src/js/lib/search.js:466]: I don't think so actually. I looked through the `ContentFile` objects in the production site and a small number start with `http://ocw.mit.edu`. None start with `https`

Comment (aliceriot) [reply src/js/lib/search.js:466]: In that case I think this is fine

**Review by aliceriot — APPROVED** [review]
lgtm! 👍

**Pull request closed without merging**